### DISCUSSION TITLE
When (if?) to synchronize beats upgrades with SO

### DISCUSSION THREAD

**Discussion by abesinger** [Security-Onion-Solutions/securityonion#6471] [Unsupported Versions] [answered]
I notice that SO 2.3.90 is has moved to using Elastic 7.15 (which is great).

I've got a number of hosts using winlogbeat to send logs. They're running v7.13, as it's the version consistent with the  Elastic running on 2.3.80. Right now it's a dozen or so hosts, eventually it will be a few hundred.

My question is, is it (or when is it?) important to coordinate upgrading winlogbeat, or other beats, when SO moves to a new version of Elastic? And if so, what's the recommended process for doing so?  Does upgrading need to be tightly synchronized -- e.g. will upgrading SO before winlogbeat break things, or vice versa?  Upgrading SO is pretty easy and low impact, upgrading the clients on a bunch of production servers is more complicated, and takes up a sysadmin's time (someone other than me), and risks breakage. I'd rather not have to ask them to install new software unless it's necessary. If tight synchronization is required, that means a lot of planning, vs. being able to upgrade clients individually over time.

**Comment by dougburks** [comment, marked as answer]
From https://docs.securityonion.net/en/2.3/soup.html#agents:
> If you’ve previously added any external agents (Wazuh, Beats, etc.), be sure to upgrade them to match the version of your upgraded components.

Older agents may continue to work, but we can't guarantee it as Elastic tends to iterate rapidly.

**Reply by abesinger** [reply]
Sounds like a definite maybe.

Let me try asking in a different way.  When testing a new release, do you test with the beats versions from the previous release?  If so, do you note anything of import about it in the release notes?

**Reply by dougburks** [reply]
No, we don't test older beat versions. If you have further questions about compatibility with older beat versions, you might want to ask the Elastic team.
